### ISSUE TITLE
`CommandOptions::help_available` set to `false` lists the command as not available instead of hiding it in v0.5

### ISSUE THREAD

**Issue by nickelc** [serenity-rs/serenity#557]
```rust
client.with_framework(
    StandardFramework::new()
        .configure(|c| c.prefix("~"))
        .command("ping", |c| c.cmd(Ping).help_available(false))
        .help(help_commands::with_embeds),
);
```

**Comment by Lakelezz** [comment]
Could you elaborate a bit?

If you set it as `help_available(false)`, it is not listed in help-commands thus hidden.

**Comment by nickelc** [comment]
The docs say `Whether command should be displayed in help list or not` but the help command lists the command as not available.

![image](https://user-images.githubusercontent.com/2128532/57790293-be897a00-773a-11e9-8959-978512820e69.png)

**Comment by Lakelezz** [comment]
I was able to reproduce it. It only happens when one command is set. Thanks for reporting.

**Comment by nickelc** [comment]
>  It only happens when one command is set.

Are sure about that because my original code has multiple command groups with multiple commands.

**Comment by Lakelezz** [comment]
Fixed via #558.
